test/integration/fake_upstream.cc

```diff
@@ -103,7 +103,21 @@ void FakeStream::encodeHeaders(const Http::HeaderMap& headers, bool end_stream)
         return;
       }
     }
-    encoder_.encodeHeaders(*headers_copy, end_stream);
+    const auto status = encoder_.encodeHeaders(*headers_copy, end_stream);
+    // simulate the connection manager's behavior to handle encoding failures.
+    if (!status.ok()) {
```

Comment:
I think this is a problem.
Is this added to fix the integration tests we had that Envoy successfully handed upstream responses with no status?  I think it's now testing the upstream works, not that Envoy works, and we need some way to force the encoder to encode this for tests, even if it looks invalid.

I think this is why coverage is now failing, because we're no longer testing handling bad responses from upstream.